Content.Shared/GameObjects/Components/Research/SharedLatheComponent.cs

```diff
@@ -0,0 +1,150 @@
+using System;
+using System.Collections.Generic;
+using Content.Shared.Materials;
+using Content.Shared.Research;
+using Mono.Cecil;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.Utility;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Shared.GameObjects.Components.Research
+{
+    public enum LatheType
```

Comment:
I'm not sure this is a good idea, personally.

I would prefer the recipes a lathe can produce be decided in a more flexible fashion. The lathe component can have recipes added/removed from it, and another component could be in charge of actually adding these.

For an autolathe this would be a static list, for a protolathe this would be handled by research levels.

Reply:
Maybe something like a LatheRecipeDatabaseComponent (name still wip) which adds the recipes to the Lathe?
It would expose a list of recipes to the serializer, so the autolathe recipe list can be defined entirely through YAML, and when actual research gets implemented, it could be inherited by another component to make it interact with research levels and stuff.

Reply:
Yeah should be fine.

Reply:
Alright then, I'll get on it tomorrow.

Reply:
Done! In the end, I called it LatheDatabase. It's a bit barebones, but I'm planning on expanding it once I start coding research.